### ISSUE TITLE
find -printf is not defined in POSIX

### ISSUE THREAD

**Issue by slycordinator** [koalaman/shellcheck#2406]
Shellcheck does not warn when one is using "find" with the GNU "-fprint" option.

#### For bugs
- My shellcheck version (`shellcheck --version` or "online"): 0.8.0

#### For new checks and feature suggestions
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
find . -type f -printf '%f\n'

```

#### Here's what shellcheck currently says:
[nothing]

#### Here's what I wanted or expected to see:
In POSIX sh, find -printf is undefined

When I run the command above in dash/bash, I get a listing of all files each on their own line as expected
When I run it in busybox, I get "find: unrecognized: -printf"

https://pubs.opengroup.org/onlinepubs/007904975/utilities/find.html

**Comment by brother** [comment]
find is an external command.

**Comment by slycordinator** [comment]
I'm aware that shellcheck is an external command.

AFAIK, shellcheck checks for arguments to some external commands to warn if they are not posix compliant.

But I could be wrong about this. Either way, it would be a nice addition to warn that the argument given is not portable.

**Comment by TinCanTech** [comment]
The comment was: "`find` is an external command" .. not `shellcheck`

In what way do you mean _not portable_ ?

**Comment by slycordinator** [comment]
I mean portable as in being easily moved from one system to another.

A script that uses "find -printf" will run perfectly fine on most Linux
distros, fail if you move it to one that uses posix utilities, and
similarly fail if you were to run it on many other Unixes.
